DelvUI/Interface/AstrologianHudWindow.cs

```diff
@@ -313,52 +292,70 @@ internal class AstrologianHudWindow : HudWindow {
                     break;
 
                 default:
-                    bar.SetText(BarTextPosition.CenterLeft, BarTextType.Custom, cardJob);
-
+                    bar.SetText(BarTextPosition.CenterLeft, BarTextType.Custom,
+                        (_config.ASTShowDrawCooldownTextBar ? cardJob : ""));
                     break;
             }
 
-            if (ShowRedrawBar) {
-                var redrawText = redrawCastInfo > 0 ? redrawCastInfo.ToString("N0") + " [" + redrawStacks.ToString("N0") + "]" : redrawStacks.ToString("N0");
+            ;
+            if (_config.ASTShowRedrawBar)
+            {
+                var redrawText = redrawCastInfo > 0
+                    ? ((!_config.ASTEnableRedrawCooldownCumulate
+                        ? ((redrawCastInfo % 30) == 0
+                            ? 30
+                            : (_config.ASTShowRedrawCooldownTextBar
+                                ? (redrawCastInfo % 30).ToString(_config.ASTEnableDecimalRedrawBar ? "N1" : "N0")
+                                : ""))
+                        : (_config.ASTShowRedrawCooldownTextBar
+                            ? redrawCastInfo.ToString(_config.ASTEnableDecimalRedrawBar ? "N1" : "N0")
+                            : "")) + (_config.ASTShowRedrawTextBar ? " [" + redrawStacks.ToString("N0") + "]" : ""))
+                    : (_config.ASTShowRedrawTextBar ? redrawStacks.ToString("N0") : "");
```

Comment:
I know nested if statements are lame but this line is really hard to read lol.
Try to clean it up a bit if possible please.